### PR TITLE
Add XML Doc snippet support

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: sdk/core/Azure.Core/src/AsyncPageable.cs
Comment (tg-msft) [inline sdk/core/Azure.Core/src/AsyncPageable.cs:18]: Any idea what will happen if we run this through our ref doc generation tools?  Might want to hold off on merging this until we're through with the release so @scbedd and @chidozieononiwu don't have any fun surprises.

File: eng/SnippetGenerator/DirectoryProcessor.cs
Comment (tg-msft) [inline eng/SnippetGenerator/DirectoryProcessor.cs:65]: We're going to dramatically increase the number of files that get scanned.  I wonder if it'd be more efficient to do it line by line?  Not something to fix now, but might be worth filing a tracking bug for.  (Doing a couple of perf measurements and speeding this up might make a good Tech Talk for the team?)

File: eng/SnippetGenerator/CSharpProcessor.cs
Comment (tg-msft) [inline eng/SnippetGenerator/CSharpProcessor.cs:13]: Nit - this should be `indent` here and in the match group below.  I was thinking that was some sort of identifier for a minute and pretty confused.

**Review by tg-msft — APPROVED** [review]
This is awesome!

File: sdk/core/Azure.Core/src/AsyncPageable.cs
Comment (pakrym) [reply sdk/core/Azure.Core/src/AsyncPageable.cs:18]: I expect something like this https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.hosting.windowsservices.webhostwindowsserviceextensions.runasservice?view=aspnetcore-3.0#examples

I'll hold of this PR until the release cycle is done.

Comment (tg-msft) [reply sdk/core/Azure.Core/src/AsyncPageable.cs:18]: Sorry, my question was more about whether a custom `snippet` attribute on the `<code>` element would cause any problems with the doc generation tools.  It _shouldn't_, but I've found previous generations of doc generation tools like Sandcastle to be very finicky.  Might be worth an extra doc generation test after we merge it as a sanity check between releases.

**Comment by weshaggard** [comment]
FYI @scbedd looks like the .NET snippet processor is also adding support for snippets in doc comments.

**Review by AlexGhiondea — APPROVED** [review]
This is cool!!!

**Pull request merged**
